### PR TITLE
Move Web PubSub back to public repo

### PR DESCRIPTION
**Pull request by srnagar**
A follow-up PR of https://github.com/Azure/azure-sdk-for-java/pull/20400 to move the webpubsub code back to public repository.

### CODE REVIEW

File: sdk/webpubsub/azure-messaging-webpubsub/README.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+requests, can also use Azure Web PubSub service.
+
+[Source code][source_code] | [Product Documentation][product_documentation] | [Samples][samples_readme]
+
+## Getting started
+
+### Prerequisites
+
+- A [Java Development Kit (JDK)][jdk_link], version 8 or later.
+- [Azure Subscription][azure_subscription]
```
Comment (juniwang) [inline sdk/webpubsub/azure-messaging-webpubsub/README.md:20]: `jdk_link` and `azure_subscription` not defined in below Links section

File: sdk/webpubsub/azure-messaging-webpubsub/README.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+do this once across all repos using our CLA.
+
+This project has adopted the [Microsoft Open Source Code of Conduct][coc]. For more information see
+the [Code of Conduct FAQ][coc_faq] or contact [opencode@microsoft.com][coc_contact] with any additional questions or
+comments.
+
+<!-- LINKS -->
+
+[source_code]: https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/webpubsub/azure-messaging-webpubsub/src
+[product_documentation]: https://docs.microsoft.com/azure/azure-signalr/signalr-overview
```
Comment (juniwang) [inline sdk/webpubsub/azure-messaging-webpubsub/README.md:169]: do we have another product page for WebPubSub? or just share the same one with SignalR?

File: sdk/webpubsub/azure-messaging-webpubsub/src/test/java/com/azure/messaging/webpubsub/WebPubSubServiceClientTests.java [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+        Response<Void> removeUserResponse = javaGroup.removeUserWithResponse("testRemoveNonExistentUserFromGroup", Context.NONE);
+        assertEquals(200, removeUserResponse.getStatusCode());
+        // assertFalse(javaGroup.checkUserExists("testRemoveNonExistentUserFromGroup"));
+    }
+
+    @Disabled
+    @Test
+    public void testAddAndRemoveUserToGroup() {
+//        WebPubSubServiceClient simpleChat = client.getHubClient("test");
+        WebPubSubGroup javaGroup = client.getGroup("java");
```
Comment (juniwang) [inline sdk/webpubsub/azure-messaging-webpubsub/src/test/java/com/azure/messaging/webpubsub/WebPubSubServiceClientTests.java:152]: Looks like this test is not completed yet. Also found codes like `SignalRHubAsyncClient` which are supposed to clean up?

File: sdk/webpubsub/azure-messaging-webpubsub/src/test/resources/session-records/WebPubSubServiceClientTests.testBroadcastBytes.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{
+  "networkCallRecords" : [ {
+    "Method" : "POST",
+    "Uri" : "https://REDACTED.webpubsubdev.azure.com/api/hubs/test/:send?excluded=%5B%5D&api-version=2021-05-01-preview",
```
Comment (juniwang) [inline sdk/webpubsub/azure-messaging-webpubsub/src/test/resources/session-records/WebPubSubServiceClientTests.testBroadcastBytes.json:4]: maybe using Prod domain `webpubsub.azure.com`?

File: sdk/webpubsub/azure-messaging-webpubsub/swagger/webpubsub.json [hunk 5]
```diff
@@ -0,0 +1,717 @@
...
+          "general"
+        ],
+        "summary": "Get service health status.",
+        "operationId": "HealthApi_GetHealthStatus",
+        "parameters": [
+          {
+            "in": "query",
+            "name": "api-version",
+            "type": "string",
+            "default": "2020-10-01"
```
Comment (juniwang) [inline sdk/webpubsub/azure-messaging-webpubsub/swagger/webpubsub.json:20]: do we have a new preview api-version for it?

File: sdk/webpubsub/test-resources.json [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        "capacity": "[parameters('capacity')]"
+      },
+      "dependsOn": [],
+      "tags": {}
+    }
+  ],
+  "outputs": {
+    "AZURE_WEBPUBSUB_CONNECTION_STRING": {
+      "type": "string",
+      "value": "[listKeys(resourceId('Microsoft.AppConfiguration/configurationStores',variables('uniqueAzConfigName')), '2019-02-01-preview').value[0].connectionString]"
```
Comment (juniwang) [inline sdk/webpubsub/test-resources.json:41]: should be `Microsoft.SignalRService/WebPubSub`?

Comment (juniwang) [reply sdk/webpubsub/test-resources.json:41]: and `uniqueAzConfigName` not defined in `variables`

File: sdk/webpubsub/azure-messaging-webpubsub/README.md (hunk 1 above)
Comment (srnagar) [reply sdk/webpubsub/azure-messaging-webpubsub/README.md:20]: Fixed.

File: sdk/webpubsub/azure-messaging-webpubsub/README.md (hunk 2 above)
Comment (srnagar) [reply sdk/webpubsub/azure-messaging-webpubsub/README.md:171]: I am not aware of another public product page for Web PubSub. @vicancy can you please let me know if there's an alternate link that can be used here?

File: sdk/webpubsub/azure-messaging-webpubsub/src/test/java/com/azure/messaging/webpubsub/WebPubSubServiceClientTests.java (hunk 3 above)
Comment (srnagar) [reply sdk/webpubsub/azure-messaging-webpubsub/src/test/java/com/azure/messaging/webpubsub/WebPubSubServiceClientTests.java:147]: This test is now removed as this requires setting up a websocket client which is out of scope for playback tests.

File: sdk/webpubsub/azure-messaging-webpubsub/swagger/webpubsub.json (hunk 5 above)
Comment (srnagar) [reply sdk/webpubsub/azure-messaging-webpubsub/swagger/webpubsub.json]: this swagger is now removed and the README now points to the api-specs swagger file.

File: sdk/webpubsub/test-resources.json (hunk 6 above)
Comment (srnagar) [reply sdk/webpubsub/test-resources.json]: Removed test-resources until the ARM template supports getting the connection string.

File: sdk/webpubsub/azure-messaging-webpubsub/src/test/resources/session-records/WebPubSubServiceClientTests.testBroadcastBytes.json (hunk 4 above)
Comment (srnagar) [reply sdk/webpubsub/azure-messaging-webpubsub/src/test/resources/session-records/WebPubSubServiceClientTests.testBroadcastBytes.json:4]: We will do another recording later when the service is available in the public Azure portal.

File: sdk/webpubsub/azure-messaging-webpubsub/README.md (hunk 2 above)
Comment (vicancy) [reply sdk/webpubsub/azure-messaging-webpubsub/README.md:171]: We can use this short link https://aka.ms/awps/doc

File: sdk/webpubsub/test-resources.json (hunk 6 above)
Comment (vicancy) [reply sdk/webpubsub/test-resources.json]: @juniwang will our ARM template supports getting connection string? What is the suggested behavior to get the connection string here? Is 
`[listKeys(resourceId('Microsoft.SignalRService/WebPubSub'), '2021-04-01-preview ').value[0].connectionString]` enough?

Comment (juniwang) [reply sdk/webpubsub/test-resources.json]: Almost: `[listKeys(resourceId('Microsoft.SignalRService/WebPubSub', 'resourceNameName'), '2021-04-01-preview').primaryConnectionString]`

File: sdk/webpubsub/azure-messaging-webpubsub/src/test/java/com/azure/messaging/webpubsub/TokenGenerationTest.java [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+        WebPubSubServiceClient client = new WebPubSubClientBuilder()
+            .hub("test")
+            // this connection string has a dummy access key for testing purposes
+            .connectionString("Endpoint=https://testendpoint.webpubsubdev.azure.com;"
+                + "AccessKey=xJItsTUmJB1m+98rVG8YepBvx5BaMnUtGtbGa/oDM+mGyZ=;Version=1.0;")
+            .buildClient();
+        WebPubSubAuthenticationToken authenticationToken = client.getAuthenticationToken(tokenOptions);
+
+        assertNotNull(authenticationToken.getAuthToken());
+        assertTrue(authenticationToken.getUrl().startsWith("ws://testendpoint.webpubsubdev.azure.com"));
```
Comment (vicancy) [inline sdk/webpubsub/azure-messaging-webpubsub/src/test/java/com/azure/messaging/webpubsub/TokenGenerationTest.java:37]: should be `wss` for `https` scheme

File: sdk/webpubsub/azure-messaging-webpubsub/src/main/java/com/azure/messaging/webpubsub/WebPubSubClientBuilder.java [hunk 8]
```diff
@@ -282,14 +282,19 @@ public WebPubSubAsyncServiceClient buildAsyncClient() {
...
 
             this.credential = new AzureKeyCredential(accessKey);
             this.endpoint = csParams.get("endpoint");
 
-            if (!csParams.containsKey("endpoint") && !csParams.containsKey("accesskey")) {
-                logger.logThrowableAsError(new IllegalArgumentException(
-                    "Connection string does not contain required 'endpoint' and 'accesskey' values"));
+            String port = csParams.get("port");
+            if (!CoreUtils.isNullOrEmpty(port)) {
+                this.endpoint = this.endpoint + ":" + port;
```
Comment (vicancy) [inline sdk/webpubsub/azure-messaging-webpubsub/src/main/java/com/azure/messaging/webpubsub/WebPubSubClientBuilder.java:297]: endpoint is a URL, it is possible that it also contains subpath, can we only override port value for the endpoint?

**Review by vicancy — APPROVED** [review]

**Pull request merged**